src/app.js

```diff
@@ -1,49 +1,66 @@
 // CAT API
 
-const express = require('express');         // imports express lib module
-const app = express();                     // express() instantiates app 
-const morgan = require('morgan');
-const bodyParser = require('body-parser')
+// QUESTION: Am I confused re. repositories?
 
-app.use(express.json());        
-app.use(bodyParser.json())      // automatically attaches parsed JSON object to req.body - should send data as raw in Postman
+const buildServer = (idGenerator) => {
 
-const catsRouter = require('./catsRouter.js');
-const breedsRouter = require('./breedsRouter.js');
+    // imports express lib module
+    // express() instantiates app 
+    const express = require('express');        
+    const app = express();      
 
-app.use('/cats', catsRouter);
-app.use('/breeds', breedsRouter);
+    // helper node modules
+    const morgan = require('morgan');                               
+    const bodyParser = require('body-parser');
 
-const PORT = 4001;
+    app.use(express.json());        
+    app.use(bodyParser.json())    
+    // Note that bodyParser attaches parsed JSON object to req.body                              
 
-app.use(morgan('tiny'));
+    // imports & class instantiation of repositories and routers
+    // note module.exports = {catClass: CatClass}
+    const breedsRouter = require('./breedsRouter');
 
-// error-handling sends back error responses 
-app.use((err, req, res, next) => {
-    let status = err.status || 500  
-    let message = err.message || 'test message'
-    res.status(status).send(message); 
-});
+    const importObject = require('./catsDb'); 
+    const CatClass = importObject.catClass; 
+    const catRepository = new CatClass(idGenerator.generate); 
 
-// sets server up (listening for reqs)
-app.listen(PORT, () => { 
-    console.log(`the server is listening for catcalls on port ${PORT}`)
-});
-module.exports = app
+    const importRouterObject = require('./catsRouter.js');      
+    const CatsRouterClass = importRouterObject.router;
+    const catsRouter = new CatsRouterClass(catRepository); 
 
+    catsRouter.initializeRoutes(); // calls the routes 
 
+    app.use('/cats', catsRouter.catsRouter);  // TypeError: Router.use() requires a middleware function but got a Object 
+    app.use('/breeds', breedsRouter);
 
+    const PORT = 4001;
+
+    app.use(morgan('tiny'));
+
+    // error-handling sends back error responses 
+    app.use((err, req, res, next) => {
+        let status = err.status || 500  
+        let message = err.message || 'test message'
```

Comment:
it's good you're thinking about fallback error messages - maybe you could replace the string with something like 'an unexpected error has occurred'?